homeassistant/components/media_player/__init__.py

```diff
@@ -44,13 +45,14 @@
 ENTITY_ID_FORMAT = DOMAIN + '.{}'
 
 ENTITY_IMAGE_URL = '/api/media_player_proxy/{0}?token={1}&cache={2}'
-ATTR_CACHE_IMAGES = 'images'
-ATTR_CACHE_URLS = 'urls'
-ATTR_CACHE_MAXSIZE = 'maxsize'
+CACHE_IMAGES = 'images'
+CACHE_MAXSIZE = 'maxsize'
+CACHE_LOCK = 'lock'
+CACHE_URL = 'url'
+CACHE_CONTENT = 'content'
 ENTITY_IMAGE_CACHE = {
-    ATTR_CACHE_IMAGES: {},
-    ATTR_CACHE_URLS: [],
-    ATTR_CACHE_MAXSIZE: 16
+    CACHE_IMAGES: collections.OrderedDict(),
```

Comment:
This sounds like an LRU cache to me… did you see https://docs.python.org/3/library/functools.html#functools.lru_cache ?

Reply:
The `lru_cache` apparently does not work with coroutines and was actually [removed when this feature was first implemented](https://github.com/home-assistant/home-assistant/pull/4063/commits/4b523e3ab815245ca3171256c6fa5561791b4f57#r85579038). So I assume that it is (still) a bad fit?